sdk/core/azure-core/CHANGELOG.md

```diff
@@ -10,6 +10,8 @@
 
 ### Other Changes
 
+- Update log level in retry policy from warning to informational [3581](https://github.com/Azure/azure-sdk-for-cpp/issues/3581).
```

Comment:
Nice CL entry!

nit: we have the issue number in the beginning of the sentence in other places

https://github.com/Azure/azure-sdk-for-cpp/blob/61264d05b3de110c925a08d78f52c14c624f0a92/sdk/core/azure-core/CHANGELOG.md#bugs-fixed-1

```suggestion
-  [[#3581]](https://github.com/Azure/azure-sdk-for-cpp/issues/3581) Update log level in retry policy from warning to informational.
```